perf: extract add-item steps 6–9 into bin/create-item (#161)
gringolito/github-backlog-management-skill: skills/add-item/SKILL.md

---

### 6. Issue Creation & Project Setup (MANDATORY)

> After validation passes, delegate all GitHub API calls to `bin/create-item`:

Author: ```suggestion
After validation passes, invoke the `create-item` Bash tool to create the issue:
```

1. Write the issue body to a temp file, e.g. `/tmp/add-item-body.md`

> 2. Write the manifest JSON to `/tmp/add-item-manifest.json`:

Author: ```suggestion
2. Write the manifest JSON file, e.g. `/tmp/add-item-manifest.json`:
```

```json
{
  "title": "<issue title>",
  "body_file": "/tmp/add-item-body.md",
  "labels": ["type:<x>", "priority:<y>", "effort:<z>"],
  "rank": <rank object — see step 7>,
  "rank_adjustments": <array — see step 7>,
  "blocked_by": [{"owner": "...", "repo": "...", "number": N}, ...],
  "blocking":   [{"owner": "...", "repo": "...", "number": N}, ...],
  "parent": <integer issue number or omit>,
  "milestone": "<milestone title or omit>"
}
```

> Omit optional fields (`rank`, `rank_adjustments`, `blocked_by`, `blocking`, `parent`, `milestone`) when not applicable.

Author: Consider moving this to a reference file (e.g. issue-manifest.md) and link it here on the issue.

> 3. Run: `bin/create-item --input /tmp/add-item-manifest.json`

Author: ```suggestion
3. Run: `create-item --input /tmp/add-item-manifest.json`
```

4. Capture the JSON blob emitted to stdout — use it for Step 10.

If `bin/create-item` exits non-zero, STOP and surface its stderr output verbatim.

---

> ### 7. Execution Rank (MANDATORY, RELATIVE)

Author: Re-order the execution so the ranking process happened before the issue creation, not after it.

- `rationale:` — per-dimension Impact / Risk / Urgency / Frequency / Dependencies
- `divergence_flag:` (if present) — the agent detected a priority/rank conflict; surface this to the user and ask them to confirm or override

Present the agent's recommendation and rationale to the user before proceeding. Normalize the output to the manifest `rank` field:
- `position: top` → `{"position": "top"}`
- `position: after_issue: 45` → `{"after_issue": 45}`
- `position: bottom` → `{"position": "bottom"}`

#### 7b. Surface re-rank suggestions for existing items

If the analysis reveals existing items that appear misranked relative to the new item OR relative to each other (e.g. a `priority:P3` sitting above a `priority:P1`), list each suggested move with rationale. DO NOT apply them silently.

#### 7c. Apply rank (USER-CONFIRMED ONLY)

After the user confirms the proposed positions, include the confirmed `rank` in the manifest and any `rank_adjustments` for re-ranked existing items. Pass them to `bin/create-item` in step 6.

If the user prefers to apply moves manually, omit `rank` and `rank_adjustments` from the manifest and instruct the user to drag-drop in the Project's web UI.

---

> ### 8. Dependencies & Sub-issue Linkage

Author: Also after the issue creation. Make sure that the steps follow a logical order.

Ask the user whether to assign this item to the active milestone:

> - If yes: include `"milestone": "<milestone-title>"` in the manifest passed to `bin/create-item`

Author: ```suggestion
- If yes: include `"milestone": "<milestone-title>"` in the manifest passed to `create-item`
```

- If no: omit the `milestone` field (will be picked up by `execute-item` only after items in the active milestone are exhausted)

---

### 10. Output

> Using the JSON blob returned by `bin/create-item`, print:

Author: ```suggestion
Using the JSON blob returned by `create-item`, print:
```
